src/plugins/intel_gpu/src/plugin/transformations/swiglu_fusion.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "swiglu_fusion.hpp"
+
+#include "intel_gpu/op/swiglu.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/swish.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+SwiGLUFusion::SwiGLUFusion() {
+    using namespace ov::pass::pattern;
+
+    // Detect SwiGLU decomposition pattern
```

Comment:
Since the kernel supports rank <= 5 only, I'd suggest to add that as a condition for fusion. Probably that could be specified inside transformation callback to have more generic pass logic (maybe someone will want to reuse that in the future)

Reply:
First of all, added it as condition for fusion